circuit-foundation/src/commonMain/kotlin/com/slack/circuit/foundation/Circuit.kt

```diff
@@ -244,11 +273,28 @@ public class Circuit private constructor(builder: Builder) {
       presenterFactories.addAll(factories)
     }
 
+    public fun setAnimatedNavDecoratorFactory(
+      decoratorFactory: AnimatedNavDecorator.Factory
+    ): Builder = apply { animatedNavDecoratorFactory = decoratorFactory }
+
+    public fun addAnimatedScreenTransform(
+      screen: KClass<out Screen>,
+      animatedNavigationTransform: AnimatedScreenTransform,
```

Comment:
I always kind of envisioned this as taking two keys (current and destination), do you think that's not necessary with this API? Example case: maybe photo viewer can do a hero element transition from a pet detail to the photo viewer but not if it's from a deeplink from home?

Reply:
Think having the single screen for a general screen override of any transition is valuable. Inside that you can check against the initial/target states to set specific screen to screen transitions. 
The API right now is definitely more verbose, so I could see a need to add some sugar on top of this to make it easier to do simpler transitions.